doc/manual/Checker_Library.md

```diff
@@ -0,0 +1,84 @@
+<!---
+This Source Code Form is subject to the terms of the Mozilla
+Public License, v. 2.0. If a copy of the MPL was not distributed
+with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Checker Library
+
+## Introduction
+
+A Checker Library is a collection of modules for one domain. Typically, it
+contains a set of checks for one standard or to test the input file for one
+specific Use Case or in one specific environment. The checks may be distributed
+in multiple Checker Bundles in the Checker Library.
+
+Each Checker Library needs to have a documentation what is included. Here we
+have to distinguish between "Checks" and "Rules". A standard specify rules,
+which are described in textual form. A check is a test implementation which
+inspects the input document to verify if it is compliant to the standard.
+
+## Rules
+
+### Rule Characteristics
+
+- Rules are atomic, easy understandable and independent of how many rules are
+  necessary in the standard.
+- A Rule should be part of a standard, as it is today in many ASAM standards.
+- Rules can come from ASAM, but also from other organizations or can be company
+  internal rules.
+- A Checker Library may also need to document rules, if they are not explicitly
+  written in the according standard. This documentation can be used as a proposal
+  to the standardization project to improve the standard itself.
+
+### Rule Documentation Template
+
+If a Checker Library needs to define and document own rules, then the following
+template can be used.
+
+**Rule ID:** _The Rule ID should not change, if there is no change in the rule._
```

Comment:
Yes, I understand... to be discussed...

Reply:
@MatteoRagni : What do you think about?

    **Rule ID:** _Unique identifier according the defined [Rule UID schema](Rule_Uid_Schema.md)_

Then we can create a second PR for your proposal.

BTW. I don't like the current wild mix of Camel / snake_case.... but that's another story.

Reply:
@andreaskern74: perfect

> BTW. I don't like the current wild mix of Camel / snake_case.... but that's another story.

yes I agree and I'm pushing for selecting one of the two and forcing all standard to stick to that. My personal preference is on snake_case (since python is tier one for the projects)

Reply:
see #7

Reply:
Agreed. snake_case without capitals. I will need to update the regex in the issue #5 , because I'm not sure I wrote them to completely prevent capitals